src/terrain/terrain.hpp

```diff
@@ -259,7 +173,7 @@ class Terrain : public voxel_utility::VoxelBase {
      */
     inline int
     pos_for_map(const NodeGroup NG) const {
-        return pos(*(NG.get_tiles().begin()));
+        return terrain_base_.pos(*(NG.get_tiles().begin()));
```

Comment:
.first()

Reply:
its a set. sets have no first

Reply:
Is it ordered? The first element of a set is the smallest.

Reply:
What are you talking about?

https://stackoverflow.com/questions/47364500/how-to-get-just-first-element-of-set-in-c

There is so much wrong here, the `.begin()` is not the number one issue.

Reply:
I'm just saying this may not be guaranteed to be deterministic.